src/transformers/pipelines/conversational.py

```diff
@@ -173,6 +190,29 @@ def __init__(self, min_length_for_response=32, *args, **kwargs):
 
         self.min_length_for_response = min_length_for_response
 
+    def _get_history(self, conversation):
+        """
+        Private function (subject to change) that simply tokenizes and concatenates past inputs. Also saves that
+        tokenization into the conversation state.
+
+        Args:
+            conversation (:class:`~transformers.Conversation`)
+
+        Returns:
+            :obj:`List[int]`: The list of tokens for the past input of that conversation.
+        """
+        history = conversation._history[:]
```

Comment:
```suggestion
        history = conversation._history
```
What do the `[:]` add?

Reply:
It forces a copy, that prevents messing up unintentionally with the original cached list before it's actually fully updated.
(If you extend without the copy, then `_history` is extended, and if there's a crash anywhere then your cache is wrong)

Reply:
Can we be explicit and say `.copy()` then? We usually try to avoid magic code :-)

Reply:
Sorry, bad habit ^_^